includes/internal/emails/class-sensei-email-sender.php

```diff
@@ -0,0 +1,200 @@
+<?php
+/**
+ * File containing the Sensei_Email_Sender class.
+ *
+ * @package sensei
+ */
+
+namespace Sensei\Internal\Emails;
+
+// phpcs:disable VariableAnalysis.CodeAnalysis.VariableAnalysis -- Prevent "Unused global variable $sensei_email_data"
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+
+require plugin_dir_path( __DIR__ ) . '../../vendor/autoload.php';
+
+use \Pelago\Emogrifier\CssInliner;
+
+/**
+ * Class Sensei_Email_Sender
+ *
+ * @package Sensei\Internal\Emails
+ */
+class Sensei_Email_Sender {
+
+	/**
+	 * Instance of class.
+	 *
+	 * @var self
+	 */
+	private static $instance;
+
+	/**
+	 * The constructor.
+	 *
+	 * @var array
+	 */
+	private function __construct() {}
+
+	/**
+	 * Fetches an instance of the class.
+	 *
+	 * @return self
+	 */
+	public static function instance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Adds all filters and actions.
+	 */
+	public function init() {
+		/**
+		 * Send email of predefined types with provided contents.
+		 *
+		 * @param string $email_type The email type.
+		 * @param array  $replacements The placeholder replacements.
+		 */
+		add_action( 'sensei_send_html_email', [ $this, 'send_email_of_type' ], 10, 2 );
+	}
+
+	/**
+	 * Send email of type.
+	 *
+	 * @param string $email_type The email type.
+	 * @param array  $replacements The placeholder replacements.
+	 *
+	 * @return WP_Post|null
+	 */
+	public function send_email_of_type( $email_type, $replacements ) {
+		$email_post = $this->get_email_post_by_type( $email_type );
+
+		if ( ! $email_post ) {
+			return;
+		}
+
+		$replacements = apply_filters( 'sensei_email_replacements', $replacements, $email_type, $email_post, $this );
+
+		$templated_output = $this->get_templated_post_content( $email_post );
+
+		$style_string = apply_filters( 'sensei_email_styles', $this->get_header_styles(), $email_type, $email_post );
+
+		$html_output_with_inlined_css = CssInliner::fromHtml( $templated_output )->inlineCss( $style_string )->render();
+
+		$subject_text = wp_strip_all_tags( $email_post->post_title );
+
+		$headers[] = 'Content-Type: text/html; charset=UTF-8';
+
+		foreach ( $replacements as $recipient => $replacement ) {
+			$email_body    = $html_output_with_inlined_css;
+			$email_subject = $subject_text;
+
+			foreach ( $replacement as $key => $value ) {
```

Comment:
That can be a way, but that'll make us run the inlining multiple times in case we have multiple emails I think, which will probably be more time-consuming